src/main/java/com/bobocode/hoverla/bring/BringApplication.java

```diff
@@ -9,18 +19,113 @@
  *
  * Usage:
  * {@code
- *     ApplicationContext = BringApplication.loadContext();
+ *     ApplicationContext = BringApplication.loadContext("com.bobocode.hoverla.bring");
  *     String bean = ApplicationContext.getBean("bean_name", String.class);
  * }
  * </pre>
+ * <p>
+ * BringApplication provides possibility to define log level. Use {@link ApplicationContextBuilder} for this purpose.
+ * <p>
+ * Usage:
+ * {@code
+ *      ApplicationContext context = BringApplication.getContextBuilder()
+ *              .logLevel(Level.DEBUG)
+ *              .build();
+ *      String bean = context.getBean("BeanName");
+ * }
+ *
+ * It is necessary to provide packages will be scan to define bean definition configs. User can pass either
+ * package names or user class that is in package which should be scanned.
  */
 public class BringApplication {
+    private BringApplication() {
+    }
+
     /**
      * Initializes and returns {@link ApplicationContext}
+     * @param packagesToScan list of packages to scan
+     */
+    public static ApplicationContext loadContext(String... packagesToScan) {
+        return getContext(packagesToScan);
+    }
+
+    /**
+     * Initializes and returns {@link ApplicationContext}
+     * @param clazz particular user class that is in the package which is candidate for scanning
```

Comment:
I guess explanation is not clear/correct and I think that having this API only confuses, rather than helps.
I'd vote for removal of such possibility and pass packages only for now.

Reply:
Also, having `@BringApplication` annotation as it was suggested is not applicable here IMO, because for instance in Spring Boot we run application with server, it runs until we stop it and we don't read beans with their API most of the time, so it's cool to have main method which is simple and automatically takes package from main class, while our API doesn't run something like server, so even if we have code with this annotation and method to which we pass the main class, user would just read beans using our API most of the time with getters and what is the reason to use it then

Reply:
@DanielChanger this functionality is removed